src/common/device_helpers.cuh

```diff
@@ -853,11 +854,13 @@ class AllReducer {
   std::vector<ncclComm_t> comms;
   std::vector<cudaStream_t> streams;
   std::vector<int> device_ordinals;
+  // use nccl one-process-per-gpu type initialization?
+  bool use_nccl_opg;
 #endif
 
  public:
-  AllReducer() : initialised_(false), allreduce_bytes_(0),
-                 allreduce_calls_(0) {}
+  AllReducer() : initialised_(false), initalised(false),
```

Comment:
You have removed the initialisation of allreduce_bytes_ and duplicated the initialisation of initalised(false).